internal/stores/objects.go

```diff
@@ -523,10 +539,45 @@ func deleteObject(tx *gorm.DB, key string) error {
 func (s *SQLStore) get(key string) (dbObject, error) {
 	var obj dbObject
 	tx := s.db.Where(&dbObject{ObjectID: key}).
-		Preload("Slabs.Slab.Shards.Contracts.Host").
+		Preload("Slabs.Slab.Shards.DBSector.Contracts.Host").
 		Take(&obj)
 	if errors.Is(tx.Error, gorm.ErrRecordNotFound) {
 		return dbObject{}, ErrObjectNotFound
 	}
 	return obj, nil
 }
+
+// TODO: Should we return slabs which are below MinShards? I think so because
+// the caller should handle that and then manually delete them if necessary.
+func (s *SQLStore) slabsForRepair(n int, failureCutoff time.Time) ([]uint, error) {
+	failureQuery := s.db.Model(&dbSlab{}).
+		Select("id as slab_id").
+		Where("last_failure < ?", failureCutoff.UTC())
+	inner := s.db.Table("(?)", failureQuery).
+		Select("slab_id, db_sector_id as sector_id").
+		Joins("LEFT JOIN shards ON slab_id = shards.db_slab_id")
+	middle := s.db.Table("(?)", inner).
+		Select("slab_id, sector_id, db_contract_id as contract_id").
+		Joins("LEFT JOIN contract_sectors ON sector_id = contract_sectors.db_sector_id")
```

Comment:
should be `NATURAL JOIN` I think

Reply:
In this case I don't think it works because the columns have different names.

Reply:
sorry you're right in this case, I blindly copy pasted the comment